src/main/java/blackjack/domain/card/CardDeck.java

```diff
@@ -0,0 +1,46 @@
+package blackjack.domain.card;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+public class CardDeck {
+
+    private static final int INIT_PROVIDING_CARD_SIZE = 2;
+
+    private final Queue<Card> cards;
+
+    private CardDeck(final Queue<Card> cards) {
+        this.cards = cards;
+    }
+
+    public static CardDeck generate() {
+        List<Card> cards = Card.cards();
+        Collections.shuffle(cards);
+        return new CardDeck(new LinkedList<>(cards));
+    }
+
+    public List<Card> provideInitCards() {
+        validateEnoughDeckSize();
+        return Arrays.asList(cards.poll(), cards.poll());
+    }
+
+    private void validateEnoughDeckSize() {
+        if (cards.size() < INIT_PROVIDING_CARD_SIZE) {
+            throw new IllegalStateException("남은 카드가 2장 미만입니다.");
```

Comment:
```suggestion
            throw new IllegalStateException("남은 카드가 " + INIT_PROVIDING_CARD_SIZE + "장 미만입니다.");
```

게임 규칙이 바뀐다면 이 메시지도 수정해줘야겠네요. 상수를 메시지에 포함시키면 수정 내용이 같이 반영되겠네요